File: test/compress_test.exs
defmodule CompressTest do
  use ExUnit.Case
  doctest Compress

  test "compresses aaabccccdd to a3bc4d2" do
    assert Compress.compress("aaabccccdd") == "a3bc4d2"
  end

  test "aaaaaffffffffffc to a5f10c" do
    assert Compress.compress("aaaaaffffffffffc") == "a5f10c"
  end

  test "compresses abcd to abcd" do
    assert Compress.compress("abcd") == "abcd"
  end

  test "compresses ccceee12eccceee to c3e4c3e3" do
    assert Compress.compress("ccceee12eccceee") == "c3e4c3e3"
  end

  test "compresses effeac01cb65c to ef2eac2bc" do
    assert Compress.compress("effeac01cb65c") == "ef2eac2bc"
  end

  test "compresses mattisagoodprogrammer to mat2isago2dprogram2er" do
    assert Compress.compress("mattisagoodprogrammer") == "mat2isago2dprogram2er"
  end
end

File: compress_test.ml
open Compress
open OUnit2

let test_case input expectation =
  input
  >:: fun _ ->
  let res = compress input in
  assert_equal expectation res


let one = test_case "aaabccccdd" "a3bc4d2"

let two = test_case "aaaaaffffffffffc" "a5f10c"

let three = test_case "abcd" "abcd"

let four = test_case "ccceee12eccceee" "c3e4c3e3"

let five = test_case "effeac01cb65c" "ef2eac2bc"

let suite = "Test Compress" >::: [ one; two; three; four; five ]

let () = run_test_tt_main suite
